FS.h tests TARGET_OS_SIMULATOR without including TargetConditionals.h, which
fails on macOS 12 where the SDK makes an undefined TARGET_OS_ macro an error
(-Werror,-Wundef-prefix=TARGET_OS_).
--- libi2pd/FS.h.orig
+++ libi2pd/FS.h
@@ -16,6 +16,9 @@
 #include <functional>
 #include <filesystem>
 #include <inttypes.h>
+#ifdef __APPLE__
+#include <TargetConditionals.h>
+#endif
 
 #ifndef STD_FILESYSTEM
 #	if (_WIN32 && __GNUG__) // MinGW GCC somehow incorrectly converts paths
